### PR TITLE
feat(efb): added company page

### PR DESCRIPTION
**Pull request by ZigTag**
Fixes [N/A]

## Summary of Changes

Allows the ability to change simbrief username in the EFB

## Screenshots (if necessary)

image

## Additional context

Discord username (if different from GitHub):

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

File: src/instruments/src/EFB/Company/Company.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+type CompanyProps = {
+    simbriefUsername: string,
+    changeSimbriefUsername: Function,
+}
+
+const Company = (props: CompanyProps) => {
+    return (
+        <div className="flex p-6 w-full">
+            <div className="w-4/12 mr-4">
+                <h1 className="text-white font-medium ml-2 mb-4 text-xl">Simbrief</h1>
```
Comment (pepperoni505) [inline src/instruments/src/EFB/Company/Company.tsx:12]: Should be SimBrief
```suggestion
                <h1 className="text-white font-medium ml-2 mb-4 text-xl">SimBrief</h1>
```

File: src/instruments/src/EFB/Company/Company.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React, {useState} from "react"
+
+type CompanyProps = {
+    simbriefUsername: string,
+    changeSimbriefUsername: Function,
```
Comment (pepperoni505) [inline src/instruments/src/EFB/Company/Company.tsx:5]:
```suggestion
    simBriefUsername: string,
    changeSimBriefUsername: Function,
```

File: src/instruments/src/EFB/Company/Company.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    return (
+        <div className="flex p-6 w-full">
+            <div className="w-4/12 mr-4">
+                <h1 className="text-white font-medium ml-2 mb-4 text-xl">SimBrief</h1>
+                <input
+                    className="text-2xl font-medium rounded-lg"
+                    type="text"
+                    placeholder="Insert Username"
+                    value={props.simbriefUsername}
+                    onChange={(event) => props.changeSimbriefUsername(event.target.value)} />
```
Comment (pepperoni505) [inline src/instruments/src/EFB/Company/Company.tsx:18]:
```suggestion
                    onChange={(event) => props.changeSimBriefUsername(event.target.value)} />
```

File: src/instruments/src/EFB/Efb.tsx [hunk 4]
```diff
@@ -204,9 +205,9 @@ class Efb extends React.Component<EfbProps, EfbState> {
         });
     }
 
-    changeSimbriefUsername = (event: React.ChangeEvent<HTMLInputElement>) => {
-        this.setState({ simbriefUsername: event.target.value.toString() });
-        window.localStorage.setItem("SimbriefUsername", event.target.value.toString());
+    changeSimbriefUsername = (name: string) => {
+        this.setState({ simbriefUsername: name });
+        window.localStorage.setItem("SimbriefUsername", name);
```
Comment (pepperoni505) [inline src/instruments/src/EFB/Efb.tsx:210]:
```suggestion
    changeSimBriefUsername = (name: string) => {
        this.setState({ simBriefUsername: name });
        window.localStorage.setItem("SimBriefUsername", name);
```

File: src/instruments/src/EFB/Efb.tsx [hunk 5]
```diff
@@ -247,7 +248,7 @@ class Efb extends React.Component<EfbProps, EfbState> {
             case 3:
                 return <h1>Page 3</h1>;
             case 4:
-                return <h1>Page 4</h1>;
+                return <Company simbriefUsername={this.state.simbriefUsername} changeSimbriefUsername={this.changeSimbriefUsername} />;
```
Comment (pepperoni505) [inline src/instruments/src/EFB/Efb.tsx:251]:
```suggestion
                return <Company simBriefUsername={this.state.simBriefUsername} changeSimBriefUsername={this.changeSimbriefUsername} />;
```

File: src/instruments/src/EFB/Efb.tsx [hunk 6]
```diff
@@ -137,7 +138,7 @@ class Efb extends React.Component<EfbProps, EfbState> {
     }
 
     fetchSimbriefUsername() {
-        const username = "myselfaaryan";
+        const username = window.localStorage.getItem("SimbriefUsername");
```
Comment (pepperoni505) [inline src/instruments/src/EFB/Efb.tsx:141]:
```suggestion
        const username = window.localStorage.getItem("SimBriefUsername");
```

**Comment by ZigTag** [comment]
@pepperoni505 Renaming all references of "Simbrief" to "SimBrief" is not in the scope of this PR

**Comment by pepperoni505** [comment]
> @pepperoni505 Renaming all references of "Simbrief" to "SimBrief" is not in the scope of this PR

Ah okay.

**Review by pepperoni505 — APPROVED** [review]

**Pull request merged**